operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,1057 @@
+    RasterResultDescriptor, SingleVectorSource, TypedRasterQueryProcessor,
+    TypedVectorQueryProcessor,
+};
+use arrow::datatypes::ArrowNativeTypeOp;
+
+use crate::error;
+use crate::processing::rasterization::GridOrDensity::Grid;
+use crate::util;
+
+use async_trait::async_trait;
+
+use futures::stream::BoxStream;
+use futures::{stream, StreamExt};
+use geoengine_datatypes::collections::GeometryCollection;
+
+use geoengine_datatypes::primitives::{
+    AxisAlignedRectangle, BoundingBox2D, Coordinate2D, Measurement, RasterQueryRectangle,
+    SpatialPartition2D, SpatialPartitioned, SpatialResolution, VectorQueryRectangle,
+};
+use geoengine_datatypes::raster::{
+    GeoTransform, Grid2D, GridOrEmpty, GridSize, GridSpaceToLinearSpace, RasterDataType,
+    RasterTile2D, TilingSpecification,
+};
+
+use num_traits::FloatConst;
+use rayon::prelude::*;
+
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use crate::util::{spawn_blocking, spawn_blocking_with_thread_pool};
+use tracing::span;
+use tracing::Level;
+use typetag::serde;
+
+/// An operator that rasterizes vector data
+pub type Rasterization = Operator<RasterizationParams, SingleVectorSource>;
+
+impl OperatorName for Rasterization {
+    const TYPE_NAME: &'static str = "Rasterization";
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum GridSizeMode {
+    /// The spatial resolution is interpreted as a fixed size in coordinate units
+    Fixed,
+    /// The spatial resolution is interpreted as a multiplier for the query pixel size
+    Relative,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+pub enum GridOrDensity {
+    /// A grid which summarizes points in cells (2D histogram)
+    Grid(GridParams),
+    /// A heatmap calculated from a gaussian density function
+    Density(DensityParams),
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DensityParams {
+    /// Defines the cutoff (as percentage of maximum density) down to which a point is taken
+    /// into account for an output pixel density value
+    cutoff: f64,
+    /// The standard deviation parameter for the gaussian function
+    stddev: f64,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GridParams {
+    /// The size of grid cells, interpreted depending on the chosen grid size mode
+    spatial_resolution: SpatialResolution,
+    /// The origin coordinate which aligns the grid bounds
+    origin_coordinate: Coordinate2D,
+    /// Determines how to interpret the grid resolution
+    grid_size_mode: GridSizeMode,
+}
+
+/// The parameter spec for `Rasterization`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RasterizationParams {
+    grid_or_density: GridOrDensity,
+}
```

Comment:
Sry, I just saw that.
You could get rid of this struct an used `GridOrDensity` directly as params.

`pub type Rasterization = Operator<GridOrDensity, SingleVectorSource>;`

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/e76f07d3aa946539f37361e631cfc725ac0bc3b2